File: go-server/domain/game.go
package domain

import (
	"time"
)

// Game schema structure
type Game struct {
	ID           string    `json:"id"`
	Name         string    `json:"name"`
	GamePack     GamePack  `json:"game_pack"`
	MinPlayers   int       `json:"min_players"`
	MaxPlayers   int       `json:"max_players"`
	Brief        string    `json:"brief"`
	Description  string    `json:"description"`
	ImgURL       string    `json:"img_url"`
	EstimateTime int       `json:"estimate_time"`
	CreatedAt    time.Time `json:"created_at"`
	UpdatedAt    time.Time `json:"updated_at"`
}

// GameRepository represent the game's repository contract
type GameRepository interface {
	FindAll() ([]*Game, error)
	FindOneByID(gamePack string) (*Game, error)
}

// GameUseCase represent the game's useacses
type GameUseCase interface {
	GetGames() ([]*Game, error)
	GetGameInfo(gamePack string) (*Game, error)
}

File: web/domain/game.ts
import { GamePack } from 'domain/Room';

export type OptionType = {
  label: string;
  value: string;
};

export type Game = {
  id: string;
  name: string;
  modes: OptionType[];
  gamePack: GamePack;
  minPlayers: number;
  maxPlayers: number;
  brief: string;
  description: string;
  imgUrl: string;
  estimateTime: number;
  createAt: string; // FIXME:
  updateAt: string; // FIXME:
};
